oci.sh

```diff
@@ -0,0 +1,205 @@
+    local rootfsdigest="$1"; shift
+    local annotations="$1"; shift
+
+    configsize="$(wc -c "$blobs/$configdigest" | awk '{print $1}')"
+    rootfssize="$(wc -c "$blobs/$rootfsdigest" | awk '{print $1}')"
+
+    content="$(jq -r -n \
+        --arg configdigest "$hash:$configdigest" \
+        --arg configsize $configsize \
+        --arg rootfsdigest "$hash:$rootfsdigest" \
+        --arg rootfssize $rootfssize \
+        --argjson annotations $annotations \
+        '{
+        schemaVersion: 2,
+        mediaType: "application/vnd.oci.image.manifest.v1+json",
+        config: {
+            mediaType: "application/vnd.oci.image.config.v1+json",
+            digest: $configdigest,
+            size: $configsize | tonumber
+        },
+        layers: [
+            {
+                mediaType: "application/vnd.oci.image.layer.v1.tar",
+                digest: $rootfsdigest,
+                size: $rootfssize | tonumber
+            }
+        ],
+        annotations: $annotations
+    } | tojson')"
+    
+    toBlob $content
+}
+
+indexdescriptor() {
+    local manifestdigest="$1"; shift
+    local os="$1"; shift
+    local arch="$1"; shift
+    local annotations="$1"; shift
+
+    manifestsize="$(wc -c "$blobs/$manifestdigest" | awk '{print $1}')"
+
+    platform=""
+
+    case "$arch" in
+    arm64v8)
+         platform=$(jq -r -n --arg os $os --arg arch "arm64" --arg variant "v8" '{
+            architecture: $arch,
+            os: $os,
+            variant: $variant
+        } | tojson')
+        ;;
+    arm32v*)
+        platform=$(jq -r -n --arg os $os --arg arch "arm" --arg variant "${arch/arm32/}" '{
+            architecture: $arch,
+            os: $os,
+            variant: $variant
+        } | tojson')
+        ;;
+    *)
+        platform=$(jq -r -n --arg os $os --arg arch $arch '{
+            architecture: $arch,
+            os: $os
+        } | tojson')
+        ;;
+    esac
+
+    jq -r -n \
+        --arg digest "$hash:$manifestdigest" \
+        --arg manifestsize $manifestsize \
+        --argjson platform $platform \
+        --argjson annotations $annotations \
+        '{
+        mediaType: "application/vnd.oci.image.manifest.v1+json",
+        digest: $digest,
+        size: $manifestsize | tonumber,
+        platform: $platform,
+        annotations: $annotations
+    } | tojson'
+}
+
+created="2025-01-21T23:32:32Z"
+
+arches=( *"/$image/hello" )
+
+manifests="[]"
+
+for path in "${arches[@]}"; do
+    os="linux"
+    arch="${path/\/hello\-world\/hello/}"
+
+    annotations=$(annotations $os $arch $created "foo")
+    rootfs=$(rootfs $os $arch)
+    config=$(config $rootfs $os $arch $created)
+    manifest=$(manifest $config $rootfs $annotations)
+    indexdescriptor=$(indexdescriptor $manifest $os $arch $annotations)
+
+    manifests=$(jq -r -n --argjson manifests $manifests --argjson manifest $indexdescriptor '$manifests  | . += [$manifest] | tojson')
+done
+
+# TODO: Windows
```

Comment:
Unfortunately, we won't be able to use OCI import for Windows (the only remotely tenable means would be foreign layers for the Windows layers, but I believe the `oci-import` builder explicitly blocks/clears those), so they'll need to stay as `Dockerfile`-based builds for the foreseeable future.